DARIAEngineering/dcaf_case_management
# Add script to safely merge daria databases

I rule and have completed some work on Case Manager that's ready for review!

We are ready to go here! The idea here is:
* we make a new fund in the instance
* we set that fund's id and that fund's current db creds as env vars
* we run something like `rails migrate_db_multitenant`

This then copies db rows over and offsets everything to preserve relations/refs, including in papertrail. It's pretty fast (I've been running it against a prod db for some stress testing and it takes roughly a minute). 

How to test drive:
* run `rails db:seed`, then run this sql to clean out your db entirely except for funds: `delete from call_list_entries; delete from events; delete from patients; delete from archived_patients; delete from configs; delete from clinics; delete from users; delete from lines; delete from calls; delete from versions; delete from notes;  delete from practical_supports; delete from external_pledges; delete from fulfillments; `
* get the first fund id with `echo "Fund.first.id" | rails c`
* `export MIGRATION_FUND_ID=(that id)`
* `export MIGRATION_DB_URL=(sandbox db url, assuming sandbox is set up to have one tenant currently)`
* `rails multitenant_db_merge`
* reset the password of test@example.com
* log in to your local and ensure it has parity with the remote db you just loaded in. it should!

This pull request makes the following changes:
* postgres migration script

no view changes

It relates to the following issue #s: 
* Bumps #1817 

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.
